drivers/i2c/i2c_dw.c

```diff
@@ -615,6 +840,14 @@ static int i2c_dw_initialize(const struct device *dev)
 	struct i2c_dw_dev_config * const dw = dev->data;
 	union ic_con_register ic_con;
 	uint32_t reg_base = get_regs(dev);
+	int ret = 0;
+
+#if defined(CONFIG_PINCTRL)
+	ret = pinctrl_apply_state(rom->pcfg, PINCTRL_STATE_DEFAULT);
+	if (ret) {
+		return ret;
+	}
```

Comment:
nit: pinctrl support is independent of slave support addition, so should probably be split into another commit

Reply:
Yes, you are right, I create another commit with pinctrl support.